### ISSUE TITLE
feat: deploy a Slack APP per environment  

### ISSUE THREAD

**Issue by rephus** [lightdash/lightdash#3968]
### Description

TL;DR Socket events are not broadcasted to all services. However  1 event is sent to 1 socket only (randomly) , independent of the URL and it could not be the one we need to unfurl. 

From Slack customer service: 

> Because you're sharing the socket connection across multiple services, the event can be sent to any of the connections, but only 1 event will be sent, and only one connection will receive it - we don't replicate them across the different connections. The documentation on this functionality can be found here:
https://api.slack.com/apis/connections/socket-implement#connections
Implementation of a distributed system can, unfortunately, be more complicated because of this. To do so you would effectively need to re-broadcast the event through some form of "distributor" that sits between the services themselves and the gateway. Having multiple "app" tokens wouldn't address this, since we would effectively be detecting them as multiple connections to the same app and treat them the same way.
Apologies for the inconvenience.

Given this limitation we can do different things: 

-  Create as many Slack app as services we need (for now:  eu1, analytics, app, and local testing) , All with the same manifest. This is my preference, as it does not affect our current implementation.
- Or We could implement a separate service (gateway or distributor) that will do the broadcasting to the right service. This  should also do the OAUTH authentication.
- Or we could hack the current implementation, to intercept the request, fake the authentication midleware, and resend the event to a new endpoint we create.

**Comment by rephus** [comment]
@owlas FYI

**Comment by rephus** [comment]
Done for analytics

**Comment by rephus** [comment]
Done for eu1 and app

**Comment by hamzahc1** [comment]
@rephus is this ticket technically in progress (looking at your last comments!)?

**Comment by TuringLovesDeathMetal** [comment]
@rephus I think we said that we'd do this "on demand" for the rest of the environments, right? 

If that's the case, can we close this issue and then open new ones as people request the feature if they're on something other than eu1 and app?

**Comment by TuringLovesDeathMetal** [comment]
^Going to close this and we'll re-open tickets as required, on demand for the environments we don't have Slack Apps for!
